lib/api-helper/goal/executeGoal.ts

```diff
@@ -380,9 +388,28 @@ export function prepareGoalExecutor(gi: GoalImplementation,
 
 export function prepareGoalInvocation(gi: GoalInvocation,
                                       listeners: GoalProjectListenerRegistration | GoalProjectListenerRegistration[]): GoalInvocation {
-    const hs: GoalProjectListenerRegistration[] =
+
+    let hs: GoalProjectListenerRegistration[] =
         listeners ? (Array.isArray(listeners) ? listeners : [listeners]) : [] as GoalProjectListenerRegistration[];
 
+    if (isLazyProjectLoader(gi.configuration.sdm.projectLoader)) {
+        // Register the materializing listener for LazyProject instances of those need to
+        // get materialized before using in goal implementations
+        const projectMaterializer = {
+            name: "clone project",
+            pushTest: AnyPush,
+            events: [GoalProjectListenerEvent.before],
+            listener: async (p: GitProject & LazyProject) => {
+                if (!p.materialized()) {
+                    // Trigger project materialization
+                    await p.materialize();
```

Comment:
Should this be wrapped in a try/catch?

Reply:
No, I want potential errors to trickle up. They will be handled in the main executeGoal function.